wit/request-reply.wit

```diff
@@ -0,0 +1,18 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
```

Comment:
Btw - Do you think this was not fulfilled with the constructs in consumer? Or is the idea mostly here just to split into its' own interface to make it an optional impl while still allow implementing `consumer.wit`?

Reply:
Yeah, this was mainly to make it an optional implementation and simplify the interface for making a request